### PR TITLE
Update the testing page

### PR DESCRIPTION
**Pull request by olemarkus**
* Use kubetest for running e2e
* A section on unit/integration tests

### CODE REVIEW

File: docs/development/testing.md [hunk 1]
```diff
@@ -1,204 +1,54 @@
...
-
-For kops-controller:
-
-```bash
-KOPS_VERSION=`bazel run //cmd/kops version -- --short`
-export DOCKER_IMAGE_PREFIX=${USER}/
-export DOCKER_REGISTRY=
-make kops-controller-push
-export KOPSCONTROLLER_IMAGE=${DOCKER_IMAGE_PREFIX}kops-controller:${KOPS_VERSION}
+go test ./...
```
Comment (hakman) [inline docs/development/testing.md:9]:
```suggestion
go test -v ./...
```

File: docs/development/testing.md [hunk 2]
```diff
@@ -1,204 +1,54 @@
...
-make kops-controller-push
-export KOPSCONTROLLER_IMAGE=${DOCKER_IMAGE_PREFIX}kops-controller:${KOPS_VERSION}
+go test -v ./...
 ```
 
-You can create a cluster using `kops create cluster <clustername> --zones us-east-1b`
+### Adding an integration test
 
-Then follow the test directions above.
+The integration tests takes a cluster spec and builds cloudformation/terraform templates. To add a new integration test, create a new directory in `tests/integration/update_cluster/` and put the cluster spec in `in-v1alpha2.yaml`. Use a unique cluster name.
```
Comment (rifelpet) [inline docs/development/testing.md:14]:
```suggestion
The integration tests takes a cluster spec and builds cloudformation/terraform templates. For new functionality, consider adding it to the `complex.example.com` cluster unless it conflicts with existing functionality in that cluster. To add a new integration test, create a new directory in `tests/integration/update_cluster/` and put the cluster spec in `in-v1alpha2.yaml`. Use a unique cluster name.
```

File: docs/development/testing.md [hunk 3]
```diff
@@ -1,204 +1,54 @@
...
 
-To override the test list for the job, you need to familiar with the
-`ginkgo.focus` and `ginkgo.skip`
-flags. Using these flags, you can do:
+Then edit `./cmd/kops/integration_test.go` and add the test function with the cluster name and directory from above.
 
-```
-export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:Performance\]"
-```
+Lastly run `hack/update-expected` to generate the expected output.
```
Comment (rifelpet) [inline docs/development/testing.md:18]:
```suggestion
Lastly run `./hack/update-expected.sh` to generate the expected output.
```

File: docs/development/testing.md [hunk 4]
```diff
@@ -1,204 +1,54 @@
...
+The container arguments for each of the jobs are the ones passed to `kubetest`.
 
-### Uploading a custom build
+Following the examples below, kubetest will download artifacts such as a given Kubernetes build. Therefore you probably want to run `kubetest` from a directory dedicated for this purpose.
 
-If you want to upload a custom Kubernetes build, here is a simple way (note:
-this assumes you've run `make quick-release` in the Kubernetes repo first):
+### Running against an existing cluster
 
+If you want `kubetest` to run spin up and tear down the cluster on every run, you can run something like the following to re-use an existing cluster.
```
Comment (rifelpet) [inline docs/development/testing.md:35]: this sentence is a bit confusing
```suggestion
You can run something like the following to have `kubetest` re-use an existing cluster.
```

File: docs/development/testing.md [hunk 5]
```diff
@@ -1,204 +1,54 @@
...
+This assumes you have already built the kops binary from source.
```
Comment (rifelpet) [inline docs/development/testing.md:36]: i think kops' `make bazel-build-cli` puts the binary in `./bazel-bin/cmd/kops/$GOOS_$GOARCH_pure_stripped/kops` so maybe its worth at least mentioning something like that.

```suggestion
This assumes you have already built the kops binary from source. The exact path to the `kops` binary used in the `--kops` flag may differ.
```

File: docs/development/testing.md [hunk 6]
```diff
@@ -1,204 +1,54 @@
...
-
-
-find kubernetes/server/bin -type f -name "*.tar" | xargs -I {} /bin/bash -c "sha1sum {} | cut -f1 -d ' ' > {}.sha1"
-find kubernetes/server/bin -type f -name "kube???" | xargs -I {} /bin/bash -c "sha1sum {} | cut -f1 -d ' ' > {}.sha1"
-
-aws s3 sync  --acl public-read kubernetes/server/bin/ s3://${S3_BUCKET_NAME}/kubernetes/dev/v1.6.0-dev/bin/linux/amd64/
-```
-
-### Example e2e command
+By adding the `--up` flag, `kubetest` will also spin up a new cluster. In most cases, you also need to add a few additional flags.
```
Comment (rifelpet) [inline docs/development/testing.md:46]:
```suggestion
By adding the `--up` flag, `kubetest` will spin up a new cluster. In most cases, you also need to add a few additional flags. See `kubetest --help 2>&1 | grep kops` for the full list.
```

**Pull request merged**